packages/components/select/src/useSelect.ts

```diff
@@ -445,10 +445,23 @@ export const useSelect = (props: ISelectProps, emit) => {
       }
     }
     if (option) return option
```

Comment:
When the options are changed, `states.cachedOptions` is empty, and this `option` is undefined.

I think we can add a backup logic to get `option` from `states.selected` before this line of code.

This might be more appropriate.

Reply:
> When the options are changed, `states.cachedOptions` is empty

Actually, once an option is selected, `cahcedOptions` shouldn't be empty.

These lines of code demonstrate that the selected options would never be deleted from cachedOptions.

https://github.com/element-plus/element-plus/blob/90c45e108d06fb7f300b5bcc3f6a325aac826c8f/packages/components/select/src/option.vue#L92-L102

However, since the data structure of `states.selected` have been normalized to Array via https://github.com/element-plus/element-plus/pull/18030, `[selected]` in line 93 will be a nested array for single selection mode,  which causes `doesSelected` to always be false. That's why the selected options are deleted from `cachedOptions` unexpectedly.

So the issue might be easy to solve with only a few changes, like this:
```diff
-     const { selected } = select.states 
-     const selectedOptions = select.props.multiple ? selected : [selected] 
+     const { selected: selectedOptions } = select.states
```

Reply:
Yeah, I didn't notice that :+1:.